### PR TITLE
Bump jenkins-pipeline-unit version

### PR DESCRIPTION
**Pull request by v1v**
- Bump version. https://github.com/jenkinsci/JenkinsPipelineUnit/releases/tag/v1.3
- Fix a broken test that used the mocked step rather than calling to the step.

### Follow-ups
- Some of the mock steps could be removed in favor of the already defined in the library.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -27,10 +27,17 @@
...
         <junit.version>4.12</junit.version>
-        <jenkins-pipeline-unit.version>1.1</jenkins-pipeline-unit.version>
+        <jenkins-pipeline-unit.version>1.3</jenkins-pipeline-unit.version>
         <groovy-eclipse-compiler.version>3.4.0-01</groovy-eclipse-compiler.version>
     </properties>
 
+    <repositories>
+        <repository>
+            <id>repo.jenkins-ci.org</id>
+            <url>https://repo.jenkins-ci.org/public/</url>
```
Comment (v1v) [inline pom.xml]: It does not exist https://repo.maven.apache.org/maven2/com/lesfurets/jenkins-pipeline-unit/1.3 yet

Comment (oleg-nenashev) [reply pom.xml]: Still not there

Comment (v1v) [reply pom.xml:37]: As long as the release is not published to the repo.maven then we need to use `repo.jenkins-ci.org`

**Comment by v1v** [comment]
any feedback?

**Comment by v1v** [comment]
@jenkins-infra/ci-maintainers , any feedback?

**Comment by v1v** [comment]
https://github.com/jenkins-infra/pipeline-library/pull/147/ was merged and bumped the version 1.4, I'll close this PR now

**Pull request closed without merging**